crates/rpc/rpc-eth-api/src/helpers/fee.rs

```diff
@@ -190,6 +255,34 @@ pub trait EthFees: LoadFee {
                     }
                 }
 
+                if let Some((pending_block, pending_receipts)) = pending_block {
+
+                    if headers.len() < block_count as usize {
```

Comment:
hmm, do we need this check here?

Reply:
yes i think, it's to prevent execeeding the user request